File: domain/src/testFixtures/java/com/study/platform/support/fake/FakeFailedNotificationRepository.java
package com.study.platform.support.fake;

import com.study.platform.domain.notification.model.FailedNotification;
import com.study.platform.domain.notification.model.FailedNotificationRepository;

import java.util.ArrayList;
import java.util.List;

public class FakeFailedNotificationRepository implements FailedNotificationRepository {

    private final List<FailedNotification> store = new ArrayList<>();

    @Override
    public FailedNotification save(FailedNotification failedNotification) {
        store.add(failedNotification);
        return failedNotification;
    }

    public List<FailedNotification> getSaved() {
        return store;
    }
}

File: domain/src/testFixtures/kotlin/com/study/platform/support/fake/FakeFailedNotificationRepository.kt
package com.study.platform.support.fake

import com.study.platform.domain.notification.model.FailedNotification
import com.study.platform.domain.notification.model.FailedNotificationRepository

class FakeFailedNotificationRepository : FailedNotificationRepository {

    private val store: MutableList<FailedNotification> = ArrayList()

    override fun save(failedNotification: FailedNotification): FailedNotification {
        store.add(failedNotification)
        return failedNotification
    }

    fun getSaved(): List<FailedNotification> = store
}
